internal/pkg/runtime/engine/apptainer/process_linux.go

```diff
@@ -848,6 +848,7 @@ func getEnvVal(env []string, envname string) string {
 // an embedded shell interpreter.
 func runActionScript(engineConfig *apptainerConfig.EngineConfig) ([]string, []string, error) {
 	args := engineConfig.OciConfig.Process.Args
+	engineConfig.OciConfig.SetProcessEnv("APPTAINER_RUNSCRIPT_TIMEOUT", engineConfig.GetRunscriptTimeout())
```

Comment:
instead of passing it to the container environment, you could directly use it to generate the context timeout here with:

```
// runscript-timeout is set, parse the duration and return an error if any,
// otherwise the timeout is set to 1 minute by default.
var execCtx context.Context
if timeoutVal := engineConfig.GetRunscriptTimeout(); timeoutVal != "" {
    timeoutDur, err := time.ParseDuration(timeoutVal)
    if err != nil {
        return nil, nil, err
    }
    execCtx = context.WithTimeout(context.Background(), timeoutDur.Abs())
} else {
    execCtx = context.WithTimeout(context.Background(), time.Minute)
}
```

then replace all `context.TODO()` in `runActionScript` by `execCtx`

Reply:
thanks! I've updated the code in this commit
https://github.com/apptainer/apptainer/pull/2116/commits/db81c41fb8117a6b696fe8d31a895c024b9135e8